staging/src/k8s.io/component-base/metrics/prometheus/controller/metrics.go

```diff
@@ -0,0 +1,151 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"k8s.io/component-base/metrics"
+	"k8s.io/component-base/metrics/legacyregistry"
+	"k8s.io/klog/v2"
+)
+
+type sinceFunc func(time.Time) time.Duration
+
+const controllerNamespace = "controller"
+
+var (
+	ReconciliationDurations = metrics.NewHistogramVec(
+		&metrics.HistogramOpts{
+			Namespace:      controllerNamespace,
+			Name:           "reconcile_time_seconds",
+			Help:           "Duration of a sync of a controller's resource",
+			StabilityLevel: metrics.ALPHA,
+			Buckets:        metrics.ExponentialBuckets(0.001, 2, 15),
+		},
+		[]string{"controller", "status"},
```

Comment:
Sorry if this has been discussed before, but would it make sense to have an instance of the metric for each subsystem, instead of having the controller as a label?